### DISCUSSION TITLE
Newbee to ESP32 audio kit v2.2 (arduino compiling errors)

### DISCUSSION THREAD

**Discussion by Zoukimando** [marcel-licence/esp32_midi_sampler#11] [General]
I'm fairly new to esp32 and arduinos, I've had some success with EPS32-cam, taking other peoples code and tweaking it, to get familiar with how it works. But so far I have had zero luck getting anything to compile for the audio board ( actually the "hello world" demo worked under esp-idf but was of no help. Arduino interface makes more sense to me). Are there any simple demos around for this board. Any help gratefully recieved

**Comment by marcel-licence** [comment]
Hm I do not have any demo in mind but for the audio kit you can use "ESP32 DEV Module" as board. In case PRAM is used it should be enabled. You could also put in compiling errors here that might help to identify your excact problem.

**Reply by Zoukimando** [reply]
Well for example, compiling your multitrack looper I get the error "'Blink_Setup' was not declared in this scope". Most things I have tried return an error for something not declared or missing. And this is not a great example for a beginner as it is so complex ( well done BTW, excellent work) its hard for me to determine if its an incompatible board, syntax or my inexperience causing issues. Hence I'm looking for some simpler starting points.

**Reply by marcel-licence** [reply]
Please ensure that all files are in the same directory when opening esp32_midi_sampler.ino. Other files will be opened automatically by arduino. Only in that case arduino can do its magic =). Hope this helps

**Reply by Zoukimando** [reply]
I've just checked and everything is in the same directory for the Multitrack Looper, and still I get the Blink_Setup error. I haven't tried the sampler, so I'll give that a go

**Reply by Zoukimando** [reply]
That says "LITTLEFS.h: No such file or directory"
I know I must be doing something wrong, I just don't know what

**Reply by marcel-licence** [reply]
It looks like some libraries are missing. Following is required:
- FS in Version 1.0
- LittleFS_esp32 in Version 1.0.6 
- SD_MMC in Version 1.0  
- WiFi in Version 1.0
- AC101 in Version 0.0.1
- SPI in Version 1.0 
- Adafruit_NeoPixel in Version 1.7.0 
- Wire in Version 1.0.1 
- Adafruit_BusIO in Version 1.7.2
